component/newstorage/couchdb/store.go

```diff
@@ -581,12 +731,59 @@ func createIndexes(db *kivik.DB, tagNamesNeedIndexCreation []string) error {
 	return nil
 }
 
+func getQueryOptions(options []newstorage.QueryOption) newstorage.QueryOptions {
+	var queryOptions newstorage.QueryOptions
+
+	for _, option := range options {
+		option(&queryOptions)
+	}
+
+	if queryOptions.PageSize == 0 {
```

Comment:
Nit: you can set defaults before calling options - that way you don't need if statements.

Reply:
Good catch. I've fixed this in #49